client/rest/src/test/java/org/elasticsearch/client/RestClientMultipleHostsTests.java

```diff
@@ -79,23 +83,28 @@ public void createRestClient() throws IOException {
             @Override
             public Future<HttpResponse> answer(InvocationOnMock invocationOnMock) throws Throwable {
                 HttpAsyncRequestProducer requestProducer = (HttpAsyncRequestProducer) invocationOnMock.getArguments()[0];
-                HttpUriRequest request = (HttpUriRequest)requestProducer.generateRequest();
-                HttpHost httpHost = requestProducer.getTarget();
+                final HttpUriRequest request = (HttpUriRequest)requestProducer.generateRequest();
+                final HttpHost httpHost = requestProducer.getTarget();
                 HttpClientContext context = (HttpClientContext) invocationOnMock.getArguments()[2];
                 assertThat(context.getAuthCache().get(httpHost), instanceOf(BasicScheme.class));
-                FutureCallback<HttpResponse> futureCallback = (FutureCallback<HttpResponse>) invocationOnMock.getArguments()[3];
+                final FutureCallback<HttpResponse> futureCallback = (FutureCallback<HttpResponse>) invocationOnMock.getArguments()[3];
                 //return the desired status code or exception depending on the path
-                if (request.getURI().getPath().equals("/soe")) {
-                    futureCallback.failed(new SocketTimeoutException(httpHost.toString()));
-                } else if (request.getURI().getPath().equals("/coe")) {
-                    futureCallback.failed(new ConnectTimeoutException(httpHost.toString()));
-                } else if (request.getURI().getPath().equals("/ioe")) {
-                    futureCallback.failed(new IOException(httpHost.toString()));
-                } else {
-                    int statusCode = Integer.parseInt(request.getURI().getPath().substring(1));
-                    StatusLine statusLine = new BasicStatusLine(new ProtocolVersion("http", 1, 1), statusCode, "");
-                    futureCallback.completed(new BasicHttpResponse(statusLine));
-                }
+                exec.execute(new Runnable() {
+                    @Override
+                    public void run() {
```

Comment:
I believe that I can't do that here because these tests are compiled with source compatibility set to 1.7.

Reply:
Oh.. 
Is there a reason why the test are compiled against 1.7 ? :confused:

Reply:
The non-test code has to be compiled against 1.7 because we want folks still on 1.7 to be able to use it. The tests kind of come along for the ride mostly to make sure that everything works properly in 1.7. For things like that it is a bit of a pain though.